api/src/entities/controllers/user.ts

```diff
@@ -88,3 +89,84 @@ export const loginUser = (req: Request, res: Response) => {
       });
     });
 };
+
+export const processOneTimeCode = (req: Request, res: Response) => {
+  let min = 100000
+  let max = 999999
+  let generatedCode = min + (Math.floor(Math.random() * (max - min)))
+  User.findOneAndUpdate({ email: req.body.email }, { oneTimeCode: generatedCode })
+    .exec()
+    .then(user => {
+      if (!user) {
+        return res.status(400).json({
+          message: 'Email is not registered',
+        })
+      }
+      const transporter = nodemailer.createTransport({
+        service: 'gmail',
+        auth: {
+          user: 'devtimr',
+          pass: envConfig.GMAIL_PASSWORD,
+        },
+      });
+      const mailOptions = {
+        from: 'devtimr@gmail.com',
+        to: req.body.email,
+        subject: 'TV Minder: OneTimeCode',
+        text: generatedCode.toString(),
+      };
+      transporter.sendMail(mailOptions, (error, info) => {
+        if (error) {
+          console.log('Nodemailer error: ', error);
+        } else {
+          res.status(200).json({ message: 'One Time Code Sent' });
+        }
+      });
+    })
+    .catch((error: Error) => {
+      return res.status(500).json({ error });
+    });
+}
+
+export const processOneTimeCodeVerification = (req: Request, res: Response) => {
+  let dt = new Date()
+  dt.setMinutes(dt.getMinutes() - 5)
```

Comment:
Hmm, can we make this 15 minutes instead of 5? Also might be useful to put it in a variable
`const TOKEN_LIFESPAN_MINS = 15`